packages/flutter_test/lib/src/window.dart

```diff
@@ -944,3 +982,263 @@ class TestPlatformDispatcher implements ui.PlatformDispatcher {
+    platformDispatcher.onMetricsChanged?.call();
+  }
+
+  /// Resets [systemGestureInsets] to the default value for this view.
+  void resetSystemGestureInsets() => systemGestureInsets = _view.systemGestureInsets;
+
+  @override
+  /// The view insets to use for this test.
+  ///
+  /// Defaults to the value provided by [ui.FlutterView.viewInsets]. This
+  /// can only be set in a test environment to emulate different view
+  /// configurations. A standard [ui.FlutterView] is not mutable from the framework.
+  ///
+  /// See also:
+  ///
+  ///   * [ui.FlutterView.viewInsets] for the standard implementation
+  ///   * [resetViewInsets] to reset this value specifically
+  ///   * [reset] to reset all test values for this view
+  ui.ViewPadding get viewInsets => _viewConfiguration.viewInsets;
+  set viewInsets(ui.ViewPadding viewInsets) {
+    _viewConfiguration = _viewConfiguration.copyWith(viewInsets: viewInsets);
+    platformDispatcher.onMetricsChanged?.call();
+  }
+
+  /// Resets [viewInsets] to the default value for this view.
+  void resetViewInsets() => viewInsets = _view.viewInsets;
+
+  @override
+  /// The device pixel ratio to use for this test.
+  ///
+  /// Defaults to the value provided by [ui.FlutterView.viewPadding]. This
+  /// can only be set in a test environment to emulate different view
+  /// configurations. A standard [ui.FlutterView] is not mutable from the framework.
+  ///
+  /// See also:
+  ///
+  ///   * [ui.FlutterView.viewPadding] for the standard implementation
+  ///   * [resetViewPadding] to reset this value specifically
+  ///   * [reset] to reset all test values for this view
+  ui.ViewPadding get viewPadding => _viewConfiguration.viewPadding;
+  set viewPadding(ui.ViewPadding viewPadding) {
+    _viewConfiguration = _viewConfiguration.copyWith(viewPadding: viewPadding);
+    platformDispatcher.onMetricsChanged?.call();
+  }
+
+  /// Resets [viewPadding] to the default value for this view.
+  void resetViewPadding() => viewPadding = _view.viewPadding;
+
+  @override
+  void render(ui.Scene scene) => _view.render(scene);
+
+  @override
+  void updateSemantics(ui.SemanticsUpdate update) {
+    _view.updateSemantics(update);
+  }
+
+  /// Resets all test values to the defaults for this view.
+  ///
+  /// See also:
+  ///
+  ///   * [resetDevicePixelRatio] to reset [devicePixelRatio] specifically
+  ///   * [resetDisplayFeatures]  to reset [displayFeatures] specifically
+  ///   * [resetPadding] to reset [padding] specifically
+  ///   * [resetPhysicalGeometry] to reset [physicalGeometry] specifically
+  ///   * [resetPhysicalSize] to reset [physicalSize] specifically
+  ///   * [resetSystemGestureInsets] to reset [systemGestureInsets] specifically
+  ///   * [resetViewInsets] to reset [viewInsets] specifically
+  ///   * [resetViewPadding] to reset [viewPadding] specifically
+  void reset() {
+    _viewConfiguration = _view.viewConfiguration.copyWith(view: this);
```

Comment:
So, the reset methods don't actually reset you to the state you were in before you overrode some properties. It gets you the view config that the underlying view has now...

Reply:
I can't shake the feeling that exposing the config on the view was a mistake. It adds soo much complexity to this and I don't see it giving us any advantages. Maybe it isn't too late to just make `configuration` private and just rely on the getter/setter.

Reply:
After messing with it in the engine repo, I'm leaning more towards getting rid of the config for both `PlatformDispatcher` and `FlutterView`. The configs don't seem to _add_ anything, except for a bunch of weird edge cases and maintenance with them needing to be kept in sync with their containing classes...

Reply:
Configs were removed per external convos and other issues that were merged in.